### ISSUE TITLE
no error on unused protected method with `reportUnusedFunction`

### ISSUE THREAD

**Issue by mozartilize** [microsoft/pyright#4951]
**Describe the bug**
no error on unused protected method with `reportUnusedFunction`

```
reportUnusedFunction [boolean or string, optional]: Generate or suppress diagnostics for
a function or method with a private name (starting with an underscore) that is not accessed.
The default value for this setting is "none".
```

so I think `_foo` should be considered as private.

**Example**

```python
class Foo:
    def _foo(self): ...  # no error

    def __bar(self): ...  # works

def _jam(): ...  # works
```

**Comment by erictraut** [comment]
This is working as designed. Methods with a single underscore are considered "protected" and are accessible by subclasses. Methods with a double underscore are considered "private".
